tests/unit/amo/components/TestRatingManager.js

```diff
@@ -377,30 +413,68 @@ describe(__filename, () => {
     });
 
     describe('submitReview', () => {
-      it('posts the reviw and dispatches the created review', () => {
+      it('posts the review and dispatches the created review', () => {
         const params = {
           rating: fakeReview.rating,
           apiState: { ...signedInApiState, token: 'new-token' },
-          addonSlug: fakeAddon.slug,
+          addonId: fakeAddon.id,
           versionId: fakeReview.version.id,
         };
 
+        const reviewResult = { ...fakeReview };
         mockApi
           .expects('submitReview')
           .withArgs(params)
-          .returns(Promise.resolve({ ...fakeReview, ...params }));
+          .returns(Promise.resolve(reviewResult));
 
         return actions.submitReview(params).then(() => {
-          sinon.assert.calledWith(dispatch, setReview(fakeReview));
+          sinon.assert.calledWith(
+            dispatch,
+            setLatestReview({
+              addonId: reviewResult.addon.id,
+              addonSlug: reviewResult.addon.slug,
+              userId: reviewResult.user.id,
+              versionId: reviewResult.version.id,
+              review: reviewResult,
+            }),
+          );
           mockApi.verify();
         });
       });
+
+      it('falls back to version ID parameter', () => {
+        const versionId = 54321;
+        const params = {
+          rating: fakeReview.rating,
+          apiState: { ...signedInApiState, token: 'new-token' },
+          addonId: fakeAddon.id,
+          versionId,
+        };
+
+        // Simulate a review for a deleted add-on version.
+        const reviewResult = { ...fakeReview, version: undefined };
```

Comment:
Should we use `denormalizeReview()`?

Reply:
This is a normalized review object, i.e. one returned from the API with all nested data structures in the JSON.